frame/democracy/src/lib.rs

```diff
@@ -1277,6 +1279,7 @@ pub mod pallet {
 				}
 			}
 
+			Self::deposit_event(Event::<T>::ProposalCanceled { prop_index });
```

Comment:
Please add a test that the event gets emitted.

Reply:
@ggwpez Ok, I added a test but the event doesn't get emitted :/, I have checked if I was getting the events the right way by testing on the ```second``` function call and it emits the event. I have also tested this on the ```blacklist``` function call and it doesn't emit the event, but it should.

I get the events like this:
```rust
fn cancel_proposal_should_work() {
  // ...
  assert_ok!(Democracy::cancel_proposal(Origin::root(), 0));
  System::events(); // <- this returns an empty array
  // ...
}
```
I am not sure if I am doing something wrong with accessing the events or if the events are actually not getting emitted.

Reply:
I dont know why its there, but the `System::set_block_number(0)` is messing up the event emission.  
Your idea is good, something like `System::assert_last_event(crate::Event::ProposalCanceled { prop_index: 0 }.into());` should then work.  
Does anyone know why its setting the block number to 0? It also works without [that line](https://github.com/paritytech/substrate/blob/3823ec4a3153a8b53a0effea3ad5a9191b46a153/frame/democracy/src/tests/public_proposals.rs#L99).